tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,479 @@
+					})
+
+					It("sample app is deleted in both clusters and the namespace is empty", func(ctx SpecContext) {
+						common.CheckNamespaceEmpty(ctx, clPrimary, "sample")
+						common.CheckNamespaceEmpty(ctx, clRemote, "sample")
+						Success("Sample app is deleted in both clusters")
+					})
+				})
+
+				When("control plane namespace and gateway are deleted in both clusters", func() {
+					BeforeEach(func() {
+						// Delete the Istio CR in both clusters
+						Expect(kubectl.Delete(controlPlaneNamespace, "istio", istioName, kubeconfig)).To(Succeed(), "Istio CR failed to be deleted")
+						Expect(kubectl.Delete(controlPlaneNamespace, "istio", istioName, kubeconfig2)).To(Succeed(), "Istio CR failed to be deleted")
+						Success("Istio CR is deleted in both clusters")
+
+						// Delete the gateway in both clusters
+						eastGatewayURL := "https://raw.githubusercontent.com/istio-ecosystem/sail-operator/main/docs/multicluster/east-west-gateway-net1.yaml"
+						Expect(kubectl.DeleteFromFile(eastGatewayURL, kubeconfig)).To(Succeed(), "Gateway deletion failed on Primary Cluster")
+
+						westGatewayURL := "https://raw.githubusercontent.com/istio-ecosystem/sail-operator/main/docs/multicluster/east-west-gateway-net2.yaml"
+						Expect(kubectl.DeleteFromFile(westGatewayURL, kubeconfig2)).To(Succeed(), "Gateway deletion failed on Remote Cluster")
+
+						// Delete the namespace in both clusters
+						Expect(kubectl.DeleteNamespace(controlPlaneNamespace, kubeconfig)).To(Succeed(), "Namespace failed to be deleted on Primary Cluster")
+						Expect(kubectl.DeleteNamespace(controlPlaneNamespace, kubeconfig2)).To(Succeed(), "Namespace failed to be deleted on Remote Cluster")
+					})
+
+					It("removes everything from the namespace", func(ctx SpecContext) {
+						Eventually(clPrimary.Get).WithArguments(ctx, kube.Key("istiod", controlPlaneNamespace), &appsv1.Deployment{}).
+							Should(ReturnNotFoundError(), "Istiod should not exist anymore")
+						Eventually(clRemote.Get).WithArguments(ctx, kube.Key("istiod", controlPlaneNamespace), &appsv1.Deployment{}).
+							Should(ReturnNotFoundError(), "Istiod should not exist anymore")
+						common.CheckNamespaceEmpty(ctx, clPrimary, controlPlaneNamespace)
+						common.CheckNamespaceEmpty(ctx, clRemote, controlPlaneNamespace)
+						Success("Namespace is empty")
+					})
+				})
+			})
+		}
+	})
+
+	AfterAll(func(ctx SpecContext) {
+		// Delete the Sail Operator from both clusters
+		Expect(kubectl.DeleteNamespace(namespace, kubeconfig)).To(Succeed(), "Namespace failed to be deleted on Primary Cluster")
+		Expect(kubectl.DeleteNamespace(namespace, kubeconfig2)).To(Succeed(), "Namespace failed to be deleted on Remote Cluster")
+
+		// Delete the intermediate CA from both clusters
+		common.CheckNamespaceEmpty(ctx, clPrimary, namespace)
+		common.CheckNamespaceEmpty(ctx, clRemote, namespace)
+	})
```

Comment:
Do we need to also delete all the resources that we're created? `Istio`, `sample`, gateways, etc.

Reply:
Sample apps are being deleted a step before finishing the node of the test:
```
When("sample apps are deleted in both clusters", func() {
					BeforeAll(func(ctx SpecContext) {
						// Delete the entire sample namespace in both clusters
						Expect(kubectl.DeleteNamespace("sample", kubeconfig)).To(Succeed(), "Namespace failed to be deleted on Primary Cluster")
						Expect(kubectl.DeleteNamespace("sample", kubeconfig2)).To(Succeed(), "Namespace failed to be deleted on Remote Cluster")
					})

					It("sample app is deleted in both clusters and the namespace is empty", func(ctx SpecContext) {
						common.CheckNamespaceEmpty(ctx, clPrimary, "sample")
						common.CheckNamespaceEmpty(ctx, clRemote, "sample")
						Success("Sample app is deleted in both clusters")
					})
				})
```

The same for Istio resources, gateways, etc